lib/response.js

```diff
@@ -46,6 +46,16 @@ function Response (req, onEnd, reject) {
 
 util.inherits(Response, http.ServerResponse)
 
+Response.prototype.setTimeout = function (msecs, callback) {
+  this.timeoutHandle = setTimeout(() => {
+    this.socket.emit('timeout')
+    if (callback) {
+      callback()
+    }
```

Comment:
I would replace this with `this.on('timeout', callback)` outside of the setTimeout callback

Reply:
done